### PR TITLE
Add documentation on features and conventions

### PR DESCRIPTION
**Pull request by kimt33**
Documentation is currently a tex file (easier at the moment for compiling equations), but can be ported over to rst later.

Includes:
1. definitions
2. cartesian to spherical transformation
3. evaluations of contractions
4. evaluations of derivatives
5. evaluations of density matrix related properties
6. multipole moment integrals
7. overlap integrals
8. integrals over differential operators
9. integrals for interaction with point charges
10. electron-electron repulsion integral

Issues:
Does not cover the "how" aspect.
Errors in equations for stress tensors and ehrenfest forces

### CODE REVIEW

**Comment by kimt33** [comment]
[Add documentation for the ordering of the contractions](https://trello.com/c/KUFcbEQS/56-add-documentation-for-the-ordering-of-the-contractions)

**Comment by kimt33** [comment]
[Add documentation on conventions and features](https://trello.com/c/WZF8RNDu/66-add-documentation-on-conventions-and-features)

File: docs/notes.tex [hunk 1]
```diff
@@ -0,0 +1,799 @@
...
+\begin{equation}
+  \label{eq:cart_prim}
+  g_i(\mathbf{r} | \mathbf{R}_A, \mathbf{a}) =
+  N(\alpha_i, \mathbf{a})
+  (x - X_A)^{a_x} (y - Y_A)^{a_y} (z - Z_A)^{a_z}
+  \exp \left(-\alpha_i |\mathbf{r} - \mathbf{R}_A|^2 \right)
+\end{equation}
+where $\mathbf{r} = (x, y, z)$, $\mathbf{R}_A$ is the center of the primitive,
+$\mathbf{a} = (a_x, a_y, a_z)$ is the Cartesian components of the angular
+momentum, $\ell = a_x + a_y + a_z$, and the
```
Comment (wilhadams) [inline docs/notes.tex:22]: remove "the"

File: docs/notes.tex [hunk 2]
```diff
@@ -0,0 +1,799 @@
...
+\begin{equation}
+  \label{eq:sph_prim_2}
+  g^s_i(r | \mathbf{R}_A, m, \ell) =
+  N_s(\alpha_i, \ell)
+  S_{\ell m} (r, \phi, \theta)
+  \exp \left(-\alpha_i r^2 \right)
+\end{equation}
+In this module, we strictly utilize Cartesian Gaussian primitives at the lower
+level and transform to the spherical form whenever needed.
+Therefore, we will denote Cartesian primitives as primitives.
```
Comment (wilhadams) [inline docs/notes.tex:86]: "primitives" will refer to Cartesian primitives

File: docs/notes.tex [hunk 3]
```diff
@@ -0,0 +1,799 @@
...
+
+\subsection{Transformation of Spherical to Cartesian}
```
Comment (wilhadams) [inline docs/notes.tex:88]: Spherical primitives to Cartesian primitives

File: docs/notes.tex [hunk 4]
```diff
@@ -0,0 +1,799 @@
...
+      }
+      \sqrt{
+        \frac{1}{(2 a_x -1)!! (2 a_y -1)!! (2 a_z -1)!!}
+      }
+    }\\
+    &= \sqrt{\frac{(2 a_x -1)!! (2 a_y -1)!! (2 a_z -1)!!}{(2\ell - 1)!!}}
+  \end{split}
+\end{equation}
+
+\section{Contraction}
```
Comment (wilhadams) [inline docs/notes.tex:148]: Contractions (plural)

File: docs/notes.tex [hunk 5]
```diff
@@ -0,0 +1,799 @@
...
+    \phi (\mathbf{r} | \mathbf{R}_A, \mathbf{a}_j, \mathbf{d}, \boldsymbol{\alpha})
+    \middle|
+    (a_j)_x + (a_j)_y + (a_j)_z = \ell
+  \right\}
+\end{equation}
+To avoid confusion with the term contraction, we use the term ``shell of
+segmented contractions''.
+
+\subsection{Generalized Contractions}
+Generalized contractions is a set of contractions that have the same center
```
Comment (wilhadams) [inline docs/notes.tex:223]: are a set

File: docs/notes.tex [hunk 6]
```diff
@@ -0,0 +1,799 @@
...
+  \phi(\mathbf{r}_n | \mathbf{R}_{A}, \mathbf{a}_j, \mathbf{d}_k, \boldsymbol{\alpha}_k)
+\end{equation}
+The returned value is an array whose rows corresponds to the basis function and
+columns corresponds to the coordinate, $\mathbf{r}_n$.
+
+These functions can be used to find the values of the orbitals at various
+points, such as a grid.
+
+\subsection{Evaluation of derivatives of contractions}
+In \verb|gbasis|, contractions can be derivatized to orbitrary orders.
```
Comment (wilhadams) [inline docs/notes.tex:392]: arbitrary

**Comment by kimt33** [comment]
Updated with the (hopefully) correct equations

**Review by wilhadams — CHANGES_REQUESTED** [review]

**Pull request merged**